sktime/forecasting/compose/_skforecast_reduce.py

```diff
@@ -0,0 +1,380 @@
+    ... )
+    >>> forecaster.fit(y_train, X=X_train)  # doctest: +SKIP
+    SkforecastAutoreg(lags=[2, 4], regressor=RandomForestRegressor())
+    >>> y_pred = forecaster.predict(fh=[1, 2, 3], X=X_test)  # doctest: +SKIP
+    >>> y_pred_int = forecaster.predict_interval(  # doctest: +SKIP
+    ...     fh=[1, 3], X=X_test, coverage=[0.6, 0.4]
+    ... )
+    >>> y_pred_qtl = forecaster.predict_quantiles(  # doctest: +SKIP
+    ...     fh=[1, 3], X=X_test, alpha=[0.01, 0.5]
+    ... )
+
+    """
+
+    _tags = {
+        "authors": ["yarnabrina"],
+        "maintainers": ["yarnabrina"],
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "enforce_index_type": [pandas.DatetimeIndex, pandas.RangeIndex],
+        "handles-missing-data": False,
+        "capability:insample": False,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": False,
+        "python_version": ">=3.8,<3.13",
+        "python_dependencies": ["skforecast>=0.11"],
+    }
+
+    def __init__(
+        self: "SkforecastAutoreg",
+        regressor: object,
+        lags: typing.Union[int, numpy.ndarray, list],
+        transformer_y: typing.Optional[object] = None,
+        transformer_exog: typing.Optional[object] = None,
+        weight_func: typing.Optional[typing.Callable] = None,
+        differentiation: typing.Optional[int] = None,
+        fit_kwargs: typing.Optional[dict] = None,
+    ) -> None:
+        self.regressor = regressor
+        self.lags = lags
+        self.transformer_y = transformer_y
+        self.transformer_exog = transformer_exog
+        self.weight_func = weight_func
+        self.differentiation = differentiation
+        self.fit_kwargs = fit_kwargs
+
+        super().__init__()
+
+        self._forecaster = None
+        self._transformer_y = None
+        self._transformer_exog = None
+
+        self._clone_estimators()
+
+    def _clone_estimators(self: "SkforecastAutoreg"):
+        """Clone the regressor and transformers."""
+        from sklearn.base import clone
+
+        self._regressor = clone(self.regressor)
+
+        if self.transformer_exog:
+            self._transformer_y = clone(self.transformer_y)
+
+        if self.transformer_exog:
+            self._transformer_exog = clone(self.transformer_exog)
+
+    def _create_forecaster(self: "SkforecastAutoreg"):
+        """Create ``skforecast.ForecasterAutoreg.ForecasterAutoreg`` model."""
+        from skforecast.ForecasterAutoreg import ForecasterAutoreg
+
+        return ForecasterAutoreg(
+            self._regressor,
+            self.lags,
+            transformer_y=self._transformer_y,
+            transformer_exog=self._transformer_exog,
+            weight_func=self.weight_func,
+            differentiation=self.differentiation,
+            fit_kwargs=self.fit_kwargs,
+        )
+
+    @staticmethod
+    def _coerce_column_names(X: pandas.DataFrame):
```

Comment:
This type hint implies `X` can only be a `pandas.DataFrame`. If you want to allow `None`, please use `typing.Optional`.